src/algorithms/search/BinarySearch.java

```diff
@@ -0,0 +1,56 @@
+package algorithms.search;
+
+import java.util.Arrays;
+
+public class BinarySearch {
+
+
+    /**
+     * This function processes the search binary algorithm using a while loop
+     * it return the index of the element if it found, otherwise it returns -1
+     * @param nums
+     * @param target
+     * @return
+     */
+    public static int find(int [] nums,int target){
+        int left = 0;
+        int right = nums.length - 1;
+
+        //we need to sort our array first, as the binary search is performed on a sorted array,
+        Arrays.sort(nums);
+
+        while (left <=right){
+            int mid = (left +right)/2;
+
+            if(target == nums[mid])
+                return mid;
+            else if(target < nums[mid])
+                right = mid-1;
+            else
+                left = mid +1;
+        }
+        return  -1;
+    }
+
+
+    /**
+     * this function perform the binary search using recursion
+     * @param nums
+     * @param target
+     * @param left
+     * @param right
+     * @return
+     */
+    public static  int recursionSearch(int [] nums,int target,int left, int right){
+        int mid = (left + right)/2;
```

Comment:
I noticed there's a potential scenario where a `StackOverflowError` might occur, which could stop the execution of the code. To ensure the robustness of the application, consider implementing proper error handling for such situations.

**Test code**

```
int items[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
System.out.println(BinarySearch.recursionSearch(items, 16, 5, 7));
```


error thrown:
![Screenshot 2024-02-08 152045](https://github.com/kambereBr/JavaDSA/assets/29512462/e5a69c49-9e49-4163-afa6-fd7e0e6a8d39)